File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        // base case, already sorted
        if (nums[0] <= nums[nums.length - 1]) {
            return nums[0];
        }

        int low = 0;
        int high = nums.length - 1;
        while (low < high) {
            int mid = low + (high - low) / 2;
            if (nums[mid] > nums[high]) {
                low = mid + 1;
            } else {
                high = mid;
            }
        }

        return nums[high];
    }
}

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        # base case
        if nums[0] < nums[-1]:
            return nums[0]
        
        l = 0
        r = len(nums) - 1

        while l < r:
            mid = l + (r - l) // 2

            if nums[mid] > nums[r]:
                l = mid + 1
            else:
                r = mid

        return nums[r]